c7n/resources/ses.py

```diff
@@ -336,3 +340,189 @@ def process(self, rulesets):
+
+
+@SESIngressEndpoint.filter_registry.register('rule-set')
+class SESIngressEndpointRuleSet(ListItemFilter):
+    """Filter for SES Ingress Endpoints to look at rule sets
+
+    The schema to supply to the attrs follows the schema here:
+     https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/mailmanager/client/get_rule_set.html
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ses-ingress-endpoint-rule-set
+                resource: ses-ingress-endpoint
+                filters:
+                - type: rule-set
+                  attrs:
+                    - type: value
+                      key: length(Actions[]|[?Archive])
+                      value: 1
+    """
+    schema = type_schema(
+        'rule-set',
+        attrs={'$ref': '#/definitions/filters_common/list_item_attrs'},
+        count={'type': 'number'},
+        count_op={'$ref': '#/definitions/filters_common/comparison_operators'}
+    )
+
+    permissions = ("ses:GetRuleSet",)
+    annotation_key = 'RuleSet'
+    annotate_items = True
+
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+        self.data['key'] = self.annotation_key
+
+    def get_item_values(self, resource):
+        if self.annotation_key not in resource:
+            client = local_session(self.manager.session_factory).client('mailmanager')
+            response = client.get_rule_set(RuleSetId=resource['RuleSetId'])
+            resource["RuleSetName"] = response["RuleSetName"]
+            resource["RuleSetArn"] = response["RuleSetArn"]
+            rules = response.get('Rules', [])
+            for rule in rules:
+                for action in rule.get("Actions", []):
+                    if "Archive" in action:
+                        target_archive = action["Archive"]["TargetArchive"]
+                        archive_details = client.get_archive(ArchiveId=target_archive)
+                        archive_details.pop("ResponseMetadata")
+                        # Convert retention period to numeric values for easier comparison
+                        action["Archive"]["TargetArchive"] = \
+                            self.convert_retention_period(archive_details)
+            resource[self.annotation_key] = rules
+
+        return resource[self.annotation_key]
+
+    def convert_retention_period(self, archive_details):
+        retention_mapping = {
+            "THREE_MONTHS": 3,
+            "SIX_MONTHS": 6,
+            "NINE_MONTHS": 9,
+            "ONE_YEAR": 12,
+            "EIGHTEEN_MONTHS": 18,
+            "TWO_YEARS": 24,
+            "THIRTY_MONTHS": 30,
+            "THREE_YEARS": 36,
+            "FOUR_YEARS": 48,
+            "FIVE_YEARS": 60,
+            "SIX_YEARS": 72,
+            "SEVEN_YEARS": 84,
+            "EIGHT_YEARS": 96,
+            "NINE_YEARS": 108,
+            "TEN_YEARS": 120,
+            "PERMANENT": 99999  # Very large value to represent "PERMANENT"
```

Comment:
Seems useful to normalize these periods into numbers. A couple things I'm wondering though...
- Is it worth making the normalized value a separate key, to preserve the original response?
- Is there any more intuitive way to reflect `PERMANENT`?
A policy example in the docstring using the period would be useful in any case.